pkg/operator/endpoint_status_builder.go

```diff
@@ -68,24 +68,155 @@ func (b *scrapeEndpointBuilder) add(target *prometheusv1.TargetsResult) error {
 	return nil
 }
 
+type ScrapeJobKind uint8
+
+const (
+	ScrapeJobInvalid ScrapeJobKind = iota
+	ScrapeJobKubelet
+	ScrapeJobPodMonitoring
+	ScrapeJobClusterPodMonitoring
+	ScrapeJobNodeMonitoring
+)
+
+func probeKindToString(kind ScrapeJobKind) string {
+	switch kind {
+	case ScrapeJobKubelet:
+		return "kubelet"
+	case ScrapeJobPodMonitoring:
+		return "PodMonitoring"
+	case ScrapeJobClusterPodMonitoring:
+		return "ClusterPodMonitoring"
+	case ScrapeJobNodeMonitoring:
+		return "NodeMonitoring"
+	default:
+		return ""
+	}
+}
+
+func parseScrapeKind(kind string) (ScrapeJobKind, error) {
+	switch kind {
+	case probeKindToString(ScrapeJobKubelet):
+		return ScrapeJobKubelet, nil
+	case probeKindToString(ScrapeJobPodMonitoring):
+		return ScrapeJobPodMonitoring, nil
+	case probeKindToString(ScrapeJobClusterPodMonitoring):
+		return ScrapeJobClusterPodMonitoring, nil
+	case probeKindToString(ScrapeJobNodeMonitoring):
+		return ScrapeJobNodeMonitoring, nil
+	default:
+		return ScrapeJobInvalid, fmt.Errorf("invalid scrape kind %q", kind)
+	}
+}
+
+func parseScrapeJob(pool string) (monitoringv1.PodMonitoringCRD, error) {
+	split := strings.Split(pool, "/")
+	kind, err := parseScrapeKind(split[0])
+	if err != nil {
+		return nil, err
+	}
+	switch kind {
+	case ScrapeJobKubelet:
+		if len(split) != 1 {
+			return nil, fmt.Errorf("invalid Kubelet scrape job format %q", pool)
+		}
+		return nil, nil
+	case ScrapeJobPodMonitoring:
```

Comment:
suggestion: Instead of hardcoding the int number, we can assign the job names to variables and calculate the value. 

In t[his file](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/main/pkg/operator/apis/monitoring/v1/pod_types.go#L71-L73) we can add 

`podKey := "PodMonitoring/%s/%s" ` in pod_tyes.go

Then in this line we can have:
`if len(split) != len(strings.Split(podKey,"/"))`
That way the code is more future proof.

Reply:
I don't expect this code to change in the future. I added a comment to explain the numbers. I felt that `strings.Split` inside a loop felt a little dirty although that may be considered a premature optimization. I'm not sure how well the Golang compiler inlines bytecode. If you're still adamant about it, I possibly add:

```golang
PodKeyFormat := "PodMonitoring/%s/%s"
PodKeySectionsAmount := strings.Split(PodKeyFormat, "/")
```

Or even something like:
```golang
// GetScrapePoolKindScope returns whether the kind is cluster scoped, namespaced or neither.
func GetScrapePoolKindScope() Scope
```

and use that to figure out how to generate and parse the keys. Thoughts?